libafl_qemu/src/emu/systemmode.rs

```diff
@@ -0,0 +1,451 @@
+use std::{
+    collections::HashMap,
+    ffi::{c_void, CStr, CString},
+    fmt::Debug,
+    mem::MaybeUninit,
+    ptr::null_mut,
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use libafl::state::{HasExecutions, State};
+use libafl_qemu_sys::{
+    libafl_load_qemu_snapshot, libafl_qemu_current_paging_id, libafl_save_qemu_snapshot,
+    qemu_cleanup, qemu_main_loop, vm_start, GuestAddr, GuestPhysAddr, GuestVirtAddr,
+};
+
+use crate::{
+    emu::{libafl_page_from_addr, IsSnapshotManager},
+    Emulator, IsEmuExitHandler, MemAccessInfo, Qemu, QemuExitReason, QemuExitReasonError,
+    QemuHelperTuple, SnapshotId, SnapshotManagerError, CPU,
+};
+
+impl SnapshotId {
+    fn get_fresh_id() -> SnapshotId {
```

Comment:
`new_unique()`?

Reply:
Or better, just `unique`?

Reply:
`gen_unique_id`?

Reply:
Well `_id` is redudant since it returns an  `..Id`, and `gen_` is somewhat implied for a constructor... But you are the qemu person so it's your API in the end :)
How about `fn gen_unique() -> Self` then?

Reply:
Sounds good